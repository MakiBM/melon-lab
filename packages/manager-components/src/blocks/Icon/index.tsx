import classNames from 'classnames';
import React, { StatelessComponent } from 'react';

export interface IconProps {
  name: string;
  iconClass?: string;
  height?: string;
  width?: string;
}

const Icon: StatelessComponent<IconProps> = ({
  name,
  height,
  width,
  iconClass,
}) => {
  const classes = classNames('icon', {
    [`${iconClass}`]: iconClass,
  });

  return (
    <svg
      className={classes}
      height={height ? height : '100%'}
      width={width ? width : '100%'}
    >
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

export default Icon;
