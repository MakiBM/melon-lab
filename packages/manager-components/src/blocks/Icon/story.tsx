import { storiesOf } from '@storybook/react';
import React from 'react';
import Icon from './index';

storiesOf('Blocks|Icon', module).add('Default', () => {
  return <Icon name="logos_default" />;
});
