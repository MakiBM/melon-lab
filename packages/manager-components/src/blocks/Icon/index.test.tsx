import React from 'react';
import Icon from './index';

describe('Icon', () => {
  const defaultElement = <Icon name="logos_default" />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "icon"', () => {
    expect(shallow(defaultElement).is('.icon')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.icon').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('');
  });
});
