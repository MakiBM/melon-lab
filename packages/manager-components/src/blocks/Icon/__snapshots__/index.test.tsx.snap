// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<svg
  className="jsx-__scopedHash icon"
  height="100%"
  width="100%"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <use
    className="jsx-__scopedHash"
    xlinkHref="#logos_default"
  />
</svg>
`;
